tasks/dataset-split/main.py

```diff
@@ -0,0 +1,149 @@
+import csv
+import logging
+import zipfile
+
+from coretex import currentTaskRun, ImageDataset, CustomDataset, SequenceDataset, ImageSample, ImageDatasetClasses, CustomSample, SequenceSample
+from coretex.networking import NetworkRequestError
+
+from utils import splitOriginalSamples, SampleType, DatasetType
+
+
+def imageDatasetSplit(splitSamples: list[list[ImageSample]], datasetClasses: ImageDatasetClasses, projectID: int) -> list[ImageDataset]:
+    splitDatasets: list[ImageDataset] = []
+
+    for index, sampleChunk in enumerate(splitSamples):
+        splitDataset = ImageDataset.createDataset(f"{currentTaskRun().id}-splitDataset-{index}", projectID)
+        splitDataset.saveClasses(datasetClasses)
+
+        for sample in sampleChunk:
+            sample.unzip()
+            addedSample = splitDataset.add(sample.imagePath)
+            logging.info(f' >> [Dataset Split] The sample "{sample.name}" has been added to the dataset "{splitDataset.name}"')
+
+            tmpAnotation = sample.load().annotation
+            if tmpAnotation is not None:
+                addedSample.saveAnnotation(tmpAnotation)
+                logging.info(f' >> [Dataset Split] The anotation for sample "{sample.name}" has been added')
+
+            try:
+                tmpMetadata = sample.loadMetadata()
+                addedSample.saveMetadata(tmpMetadata)
+                logging.info(f' >> [Dataset Split] The metadata for sample "{sample.name}" has been added')
+            except FileNotFoundError:
+                logging.info(f' >> [Dataset Split] The metadata for sample "{sample.name}" was not found')
+            except ValueError:
+                logging.info(f' >> [Dataset Split] Invalid metadata type for sample "{sample.name}"')
+        
+        splitDatasets.append(splitDataset)
+
+        logging.info(f' >> [Dataset Split] New dataset named "{splitDataset.name}" contains {len(sampleChunk)} samples')
+
+    return splitDatasets
+
+
+def customDatasetSplit(splitSamples: list[list[CustomSample]], projectID: int) -> list[CustomDataset]:  
+    splitDatasets: list[CustomDataset] = []
+
+    for index, sampleChunk in enumerate(splitSamples):
+        splitDataset = CustomDataset.createDataset(f"{currentTaskRun().id}-splitDataset-{index}", projectID)
+
+        for sample in sampleChunk:
+            sample.unzip()
+            splitDataset.add(sample.zipPath)
+            logging.info(f' >> [Dataset Split] The sample "{sample.name}" has been added to the dataset "{splitDataset.name}"')
+
+        splitDatasets.append(splitDataset)
+
+        logging.info(f' >> [Dataset Split] New dataset named "{splitDataset.name}" contains {len(sampleChunk)} samples')
+
+    return splitDatasets
+
+
+def sequenceDatasetSplit(splitSamples: list[list[SequenceSample]], metadata: CustomSample, projectID: int) -> list[CustomDataset]:
+    metadataAddress = list(metadata.load().folderContent)[0] #address where the file metadata is located in the form of a string
```

Comment:
Python comments on the same line as the code should be separated with 2 spaces, not one. Also add space after the `#` as well.